src/types/schema/apisix/consumers.ts

```diff
@@ -24,7 +24,8 @@ const Consumer = z
     username: z
       .string()
       .min(1)
-      .regex(/^[a-zA-Z0-9_]+$/),
+      // ref: https://github.com/apache/apisix/blob/a2482df74d712228a1a6644662d74d2f51a3f5e6/apisix/schema_def.lua#L713
+      .regex(/^[a-zA-Z0-9_-]+$/),
```

Comment:
This method is equivalent to apisix.

<img width="819" height="1250" alt="Image" src="https://github.com/user-attachments/assets/a28ef2a3-31cb-4a06-b9ea-62a1835ad296" />